File: src/main/java/awktal/mule/SceneManager.java
package awktal.mule;

import awktal.mule.GameScene;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Stage;

public class SceneManager {
    private static Stage stage;
    private static GameState state;

    /**
     * Initializes the scene manager with the stage that it will control.
     * @param stage the stage that the scene manager will show screens on.
     * @param state the game state to use.
    */
    public static void initSceneManager(Stage stage, GameState state) {
        SceneManager.stage = stage;
        SceneManager.state = state;
        SceneController.setGameState(SceneManager.state);
        TurnManager.setGameState(SceneManager.state);
    }

    /**
     * Loads a new scene onto the stage.
     * @param gameScene the gameScene to be loaded.
     * @return the controller that was given control.
    */
    public static SceneController loadScene(GameScene gameScene) {
        SceneController controller = null;
        try {
            FXMLLoader loader = new FXMLLoader(SceneManager.class.getResource(gameScene.getFxml()));
            Parent screen = loader.load();
            controller = (SceneController) loader.getController();
            Scene scene = new Scene(screen, stage.getWidth(), stage.getHeight());
            stage.setScene(scene);
        } catch (Exception e) {
            e.printStackTrace();
            throw new RuntimeException("Cannot load scene " + gameScene.toString()
                + ". Check if file " + gameScene.getFxml() + "exists. \n" + " error: "
                + e.toString());
        }
        return controller;
    }
}

File: web/current/web/SceneManager.dart
import 'dart:html';

import 'GameScene.dart';
import 'GameState.dart';
import 'ControllerFactory.dart';

class SceneManager {

    static DivElement shadow_stage = null;

    static void loadScene(GameScene scene, GameState gameState) {
        if (shadow_stage == null) {
            DivElement stage = querySelector("#stage");
            shadow_stage = new DivElement();
            shadow_stage.id = "shadow_stage";
            stage.children.add(shadow_stage);
        }
        var controller = ControllerFactory.newController(scene, shadow_stage, gameState);
    }
}